Gems/Terrain/Code/Tests/TerrainSystemBenchmarks.cpp

```diff
@@ -322,6 +322,36 @@ namespace UnitTest
         ->Args({ 4096, 1, static_cast<int>(AzFramework::Terrain::TerrainDataRequests::Sampler::EXACT) })
         ->Unit(::benchmark::kMillisecond);
 
+    BENCHMARK_DEFINE_F(TerrainSystemBenchmarkFixture, BM_ProcessHeightsRegion)(benchmark::State& state)
```

Comment:
I added the benchmarks for the ProcessRegion functions first. What's the best approach for the ProcessList functions? I was thinking of having a separate function called during InternalSetup() that would populate a vector with a bunch of locations. Since if we do that in RunTerrainApiBenchmark() it would end up counting the time taken to populate the input list, which isn't something we'd want to do.

Reply:
Ok. I'll make it part of the benchmark run for the ProcessList API when I add that.

Reply:
I thought about this some more. Is there a use case where I could have a container with a bunch of input positions cached which I use every frame to call ProcessList functions? In that scenario, whichever module has the cached input positions only has a one time cost of setting up that container with input positions. All subsequent frames will pass in that container to the ProcessList functions. In that case, it would be useful to only measure the call to ProcessList in the benchmark and not the input positions setup. Not sure if this is just a contrived example though. If we needed to re-populate/re-create a container with input positions each frame and then loop over those positions again(in ProcessList function), wouldn't it be faster to call the Get* functions directly(this depends on what the final optimized version of ProcessList functions do, so I may be wrong here)?

Reply:
I was wrong. Even with generating inputs as part of the benchmark, the ProcessList functions are faster than the calls to the individual Get* functions.

Reply:
Yeah, from what I've seen in the Gradient APIs, just eliminating the general EBus per-call overhead saves anywhere from 10-50% all by itself.  For the container population, assuming you know the size in advance (which you will), you should get a single allocation hit from a reserve() call, but then every assignment shouldn't be much different than doing "AZ::Vector3 position = AZ::Vector3(x, y, z)" for every individual Get() call anyways.  So as long as you're calling enough points to swallow the single allocation, keeping the container population in the benchmark should be negligible for the results.  I just don't want to "cheat" the results by doing actual less work for the Process* functions than the equivalent Get* ones.  Guess I could accept it if we wanted to set up the same list of points for both outside the profile, but since callers will generally need to do that work, it's kinda nice to keep it in the benchmark, IMO.

As for the use case, it generally _shouldn't_ be a valid use case to call ProcessList with the same cached positions every frame, because that ought to produce the same outputs every frame and would be a waste, the caller ought to cache the outputs at that point too.  You'd only want to re-call if you got an OnTerrainDataChanged notification.  I guess if you wrote some sort of dynamic terrain that was changing every frame then it might happen, but that seems pretty fringe.